### PR TITLE
delete unchanged previous files

### PR DESCRIPTION
**Pull request by Fabio1988**
As there is now a daily sync it's annoying to create a daily commit if nothing changed. therefore I added a comparison of content, if it's unchanged we delete old file again. if the newly created directory `previous-versions/$sha` is empty it should not be commited and therefor the workflow does not create a commit :+1:

### CODE REVIEW

File: generate.js [hunk 1]
```diff
@@ -65,6 +66,18 @@ const masterfile = async function update() {
       }
       fs.writeFileSync(`./${templateName}`, JSON.stringify(newData, null, 2), 'utf8', () => { })
 
+      // compare content of both files
+      var newFile = fs.readFileSync(`./${templateName}`)
+      var previousFile = fs.readFileSync(previousFilePath)
+      if (previousFile.equals(newFile)) {
```
Comment (Fabio1988) [inline generate.js:72]: https://nodejs.org/api/all.html#buffer_buf_equals_otherbuffer

File: generate.js [hunk 2]
```diff
@@ -65,6 +66,18 @@ const masterfile = async function update() {
       }
       fs.writeFileSync(`./${templateName}`, JSON.stringify(newData, null, 2), 'utf8', () => { })
 
+      // compare content of both files
+      var newFile = fs.readFileSync(`./${templateName}`)
```
Comment (TurtIeSocks) [inline generate.js]:
```suggestion
      const newFile = fs.readFileSync(`./${templateName}`)
```
Please change these to `const`

File: generate.js [hunk 3]
```diff
@@ -65,6 +66,18 @@ const masterfile = async function update() {
...
+      var previousFile = fs.readFileSync(previousFilePath)
```
Comment (TurtIeSocks) [inline generate.js]:
```suggestion
      const previousFile = fs.readFileSync(previousFilePath)
```

**Review by TurtIeSocks — CHANGES_REQUESTED** [review]
Small, minor changes, everything works great though! Thanks so much, this will be awesome to have!

**Review by TurtIeSocks — APPROVED** [review]

**Pull request merged**